src/pt/mangalivre/src/eu/kanade/tachiyomi/extension/pt/mangalivre/MangaLivre.kt

```diff
@@ -1,17 +1,118 @@
 package eu.kanade.tachiyomi.extension.pt.mangalivre
 
+import android.app.Application
+import android.content.SharedPreferences
+import android.widget.Toast
+import androidx.preference.EditTextPreference
+import androidx.preference.PreferenceScreen
 import eu.kanade.tachiyomi.multisrc.madara.Madara
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import okhttp3.FormBody
+import okhttp3.Response
+import org.json.JSONArray
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.io.IOException
 import java.text.SimpleDateFormat
 import java.util.Locale
 
-class MangaLivre : Madara(
-    "Manga Livre",
-    "https://mangalivre.tv",
-    "pt-BR",
-    SimpleDateFormat("MMMM dd, yyyy", Locale("pt")),
-) {
+class MangaLivre :
+    Madara(
+        "Manga Livre",
+        "https://mangalivre.tv",
+        "pt-BR",
+        SimpleDateFormat("dd.MM.yyyy", Locale("pt")),
+    ),
+    ConfigurableSource {
 
-    override val chapterUrlSuffix: String = ""
-    override val useNewChapterEndpoint = true
     override val id: Long = 2834885536325274328
+    override val useLoadMoreRequest = LoadMoreStrategy.Always
+    override val baseUrl by lazy { getPrefBaseUrl() }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override val client = super.client.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Origin", baseUrl)
+        .set("X-Requested-With", "XMLHttpRequest")
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val html = response.body.string()
+        var document = Jsoup.parse(html)
+
+        if (document.selectFirst("li.wp-manga-chapter") == null) {
+            val mangaUrl = response.request.url.toString().trimEnd('/')
+            val headers = headersBuilder().set("Referer", mangaUrl).build()
```

Comment:
You shouldn't need to send the full Referer header. In browsers it is possible to set it up to just send the domain itself.
```suggestion
            val headers = headersBuilder().set("Referer", "$baseUrl/").build()
```